docs/user_guide/secure_payloads.rst

```diff
@@ -0,0 +1,100 @@
+Secure Payloads
+================
+
+.. warning::
+    This page is still under development and not complete. It will be so until
+    this warning is removed.
+
+Secure payloads offer the ability to securely provision secrets to an enrolled node.
+
+The payload itself is sent via the Keylime Tenant to the verifier and only when a node
+has passed its enrolment criteria (including any `tpm_policy` or IMA whitelist),
+will the `v_key` half be passed to the agent to decrypt the payload.
+
+The follow flow will occur during execution of a Secure payload:
+
+* The tenant will first set up a folder containing the files to be copied and the scripts to be executed by the agent
+* The tenant will generate a new CA (if one hasn't been generated yet) and then generate a new cert/key combo for the agent to be bootstrapped
+* The tenant will include the payload files with the `--include` option
+* Tenant and verifier will bootstrap the agent and derive a key as normal
+* Once the bootstrap key has been derived and tpm_policy and IMA verification has passed, Keylime will decrypt and extract the zipped payload that includes both the cert/key and files situated in the payload folder.
+* Then Keylime will run the provided autorun.sh script.
+
+The scripts work by leveraging two features of Keylime:
+
+* Automatic certificate generation/delivery
+* The ability to run scripts provided in the payload after successful bootstrapping
+
+
+For the following example, we will provision some SSH keys onto the agent.
+
+1. Create a directory to host the files and `autorun.sh` script. For this example, we will use `payload`
+2. Create an `autorun.sh` script
+
+```
+#!/bin/bash
+
+mkdir -p /root/.ssh/
+cp payload_id_rsa* /root/.ssh/
+chmod 600 /root/.ssh/payload_id_rsa*
+```
+
+3. Copy the files you wish to provision into the `payload` directory.
+
+```
+$ ls payload/
+payload_id_rsa.pub
+payload_id_rsa
+```
+
+Send the files using the Keylime Tenant
+
+```
+keylime_tenant -t <agent-ip> --cert myca --include payload
```

Comment:
Maybe we should change `myca` to just the default location (`/var/lib/keylime/ca`) to avoid confusion here?